dvc/fs/path.py

```diff
@@ -15,7 +15,7 @@ def join(self, *parts):
         return self.flavour.join(*parts)
 
     def parts(self, path):
-        drive, path = self.flavour.splitdrive(path)
+        drive, path = self.flavour.splitdrive(path.rstrip(self.flavour.sep))
```

Comment:
Could you please add a simple unit test for this? (e.g. create tests/unit/fs/test_path.py for it) This module is kinda experimental, but since we are fixing such things already, it would be great to keep track of them.